thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/NotificationOnboardingTask.java

```diff
@@ -71,20 +86,50 @@ public void run() {
     DateTime start = (DateTime) executionContext.getExecutionResult(NOTIFICATION_START);
     DateTime end = (DateTime) executionContext.getExecutionResult(NOTIFICATION_END);
 
-
+    // Get alert config from previous output
     AlertConfigDTO alertConfig = (AlertConfigDTO) executionContext.getExecutionResult(ALERT_CONFIG);
 
     SmtpConfiguration smtpConfiguration = new SmtpConfiguration();
     smtpConfiguration.setSmtpHost(taskConfigs.getString(SMTP_HOST));
     smtpConfiguration.setSmtpPort(taskConfigs.getInt(SMTP_PORT));
-    EmailResource emailResource = new EmailResource(smtpConfiguration, alertFilterFactory,
-        taskConfigs.getString(DEFAULT_ALERT_SENDER_ADDRESS), taskConfigs.getString(DEFAULT_ALERT_RECEIVER_ADDRESS),
-        taskConfigs.getString(THIRDEYE_DASHBOARD_HOST), taskConfigs.getString(PHANTON_JS_PATH), taskConfigs.getString(ROOT_DIR));
+
+    MergedAnomalyResultManager mergeAnomalyDAO = DAORegistry.getInstance().getMergedAnomalyResultDAO();
+    List<MergedAnomalyResultDTO> anomalyCandidates = mergeAnomalyDAO
+        .findByStartTimeInRangeAndFunctionId(start.getMillis(), end.getMillis(), functionId, true);
+    anomalyCandidates = AlertFilterHelper.applyFiltrationRule(anomalyCandidates, alertFilterFactory);
+    List<AnomalyResult> filteredAnomalyResults = new ArrayList<>();
```

Comment:
Hi Yung-Yu,  since here anomalyCandidates is a list with filtration rule being applied already, what's the purpose to use another list "filteredAnomalyResults" to store the same information?

Reply:
It's because of the interface. The content formatter takes a Collection of AnomalyResult. I cannot directly apply List<MergedAnomalyResultDTO>. I need to cast the MergedAnomalyResultDTO to AnomalyResult to fit the input argument.